examples/router_resource.rs

```diff
@@ -0,0 +1,65 @@
+//! Example: Updating components with use_resource
+//! -----------------
+//!
+//! This example shows how to use use_reactive to update a component properly
+//! when linking to it from the same component, when using use_resource
+
+use dioxus::prelude::*;
+
+#[derive(Clone, Routable, Debug, PartialEq)]
+enum Route {
+    #[route("/")]
+    Home {},
+    #[route("/blog/:id")]
+    Blog { id: i32 },
+}
+
+fn main() {
+    launch(App);
+}
+
+#[component]
+fn App() -> Element {
+    rsx! {
+        Router::<Route> {}
+    }
+}
+
+#[component]
+fn Blog(id: i32) -> Element {
```

Comment:
This is a really good use case for `ReadOnlySignal<i32>`. We automatically convert `T` to `ReadOnlySignal<T>` in the props so it will work out of the box with the router and it will be reactive without the extra `use_reactive` hook

Reply:
use_reactive is good for non-reactive state that is created in the middle of a component like this:
```rust
let state = use_signal(|| 0);
let doubled = state() * 2;
use_effect(use_reactive((&doubled,), |(doubled,)| println!("Manually manipulate the dom")));
```

Reply:
Ok, that makes sense, however 2 questions then:
1. Is that distinction documented somewhere? After this PR gets merged (in some form), I already intend to update https://dioxuslabs.com/learn/0.5/reference/use_resource with a link to the example and an explanation
2. If it is already done with props, so should it happen with the `#[component]` approach as well? As in, should this just work without the `use_reactive` or explicit `ReadOnlySignal` at all?

Reply:
> 1. Is that distinction documented somewhere? After this PR gets merged (in some form), I already intend to update https://dioxuslabs.com/learn/0.5/reference/use_resource with a link to the example and an explanation

Not yet

> 2. If it is already done with props, so should it happen with the `#[component]` approach as well? As in, should this just work without the `use_reactive` or explicit `ReadOnlySignal` at all?

The conversion happens when you use the component if you explicitly use `ReadOnlySignal<T>` in your props or `#[component]`.

```rust
// Because value is a signal, it is reactive and any reads will automatically add the signal as a dependency to use_effect, use_resource, etc
#[component]
fn Comp(value: ReadOnlySignal<i32>) -> Element {None}

Comp {
   // A new signal is created here
   value: 0
}
```

Reply:
I will update the example, and also add documentation around this.